Automattic/jetpack
# New Package: Promote Posts

Introduces a new functionality package: Promote Posts.
>  "Attract high-quality traffic to your site using Promoted Posts."

https://user-images.githubusercontent.com/7129409/207340836-5fd1670a-75ce-4885-b8cd-0f00580bc1bd.mov


#### Changes proposed in this Pull Request:

New package with base functionality. 

#### Jetpack product discussion

Lots more context here:

- pedfti-DJ-p2#comment-849
- p1HpG7-jCm-p2
- 
#### Does this pull request change what data or activity we track or use?


Not unless you count the Redirect. 

#### Testing instructions:

Mostly code review needed. This only introduces the Package. Meant to be iterated on in future PRs.
For now, the functionality can be tested by: 
- Add to Jetpack's composer.json
- Add to Social's composer.json
- I'm not sure if there's a better way, but I ended up running `tools/check-intra-monorepo-deps.sh -vau` to get the new package linked up right 
- Put this in module-extras.php (or a place of your liking)
```
use Automattic\Jetpack\Promote_Posts;
Promote_Posts::configure();
```
- Do what's being done in the video. 
- - Jetpack plugin
- - Social plugin
- Ensure redirects are working.
